packages/admin-ui/src/button/button.tsx

```diff
@@ -1,110 +1,96 @@
-import type { ReactNode } from 'react'
-import React from 'react'
-import { Button as ReakitButton } from 'reakit/Button'
-import type { VariantProps } from '@vtex/admin-ui-core'
-import {
-  createComponent,
-  createHook,
-  useElement,
-  IconContainer,
-} from '@vtex/admin-ui-react'
+import type { ComponentPropsWithoutRef, ReactNode } from 'react'
+import React, { forwardRef } from 'react'
+
+import { IconContainer } from '@vtex/admin-ui-react'
 
 import { Center } from '../center'
 import { Spinner } from '../spinner'
-import * as style from './button.style'
+import {
+  buttonTheme,
+  innerContainerTheme,
+  spinnerContainerTheme,
+} from './button.css'
+import { cx } from '@vtex/admin-ui-core'
 
 /**
  * Button behavior
  * @example
  * const buttonProps = useButton({})
  */
-export const useButton = createHook<typeof ReakitButton, ButtonOptions>(
-  ({
-    icon,
-    size = 'normal',
-    variant = 'primary',
-    iconPosition: defaultIconPosition = 'start',
-    loading = false,
-    disabled = false,
-    bleedY = false,
-    bleedX = false,
-    children,
-    ...props
-  }) => {
-    const iconOnly = icon && !children
+export const useButton = ({
+  icon,
+  size = 'normal',
+  variant = 'primary',
+  iconPosition: defaultIconPosition = 'start',
+  loading = false,
+  disabled = false,
+  bleedY = false,
+  bleedX = false,
+  className = '',
+  children,
+  ...props
+}: ButtonProps) => {
```

Comment:
```suggestion
export const useButton = (props: ButtonProps) => {

const {
  icon,
  size = 'normal',
  variant = 'primary',
  iconPosition: defaultIconPosition = 'start',
  loading = false,
  disabled = false,
  bleedY = false,
  bleedX = false,
  className = '',
  children,
  ...htmlProps
} = props
```

Reply:
Should we keep the `useComponent` hook pattern in the new component model?

Reply:
We didn't discuss that, I don't think it was a rule on the former model either. It could easily be changed and the hooks content integrated on the component. One advantage I see for keeping the hook for the button specifically is that it could enable someone to use our button stylings with a different element than "button", since we won't have the "as" prop anymore. Someone could use this hook to generate the props and plug them on a different element

Reply:
>  I don't think it was a rule on the former model either.

That's true, maybe we can define it on the new component model. IMO it can be useful as well